### PR TITLE
Update to cosign-2.2.0, pipeline-0.51.0, go-1.20

### LINKED ISSUE #918: Issue with cosign version 2.2.0. Controller unable to sign with error: error configuring x509 signer: decrypt: encrypted: unexpected kdf
# Expected Behavior
Follow the tutorial: https://tekton.dev/docs/chains/getting-started-tutorial/
No issues are expected

# Actual Behavior
Observing issues in controller:
```
{"level":"info","ts":"2023-09-07T18:32:20.217Z","logger":"watcher","caller":"x509/x509.go:195","msg":"Found cosign key...","knative.dev/controller":"github.com.tektoncd.chains.pkg.reconciler.taskrun.Reconciler","knative.dev/kind":"tekton.dev.TaskRun","knative.dev/traceid":"5111d2fc-3526-443f-ae5a-a1e630098a3a","knative.dev/key":"edp-delivery-sk-dev/goapp-main-build-luxw-kaniko-build"}

{"level":"warn","ts":"2023-09-07T18:32:20.217Z","logger":"watcher","caller":"chains/signing.go:66","msg":"error configuring x509 signer: decrypt: encrypted: unexpected kdf parameters","knative.dev/controller":"github.com.tektoncd.chains.pkg.reconciler.taskrun.Reconciler","knative.dev/kind":"tekton.dev.TaskRun","knative.dev/traceid":"5111d2fc-3526-443f-ae5a-a1e630098a3a","knative.dev/key":"edp-delivery-sk-dev/goapp-main-build-luxw-kaniko-build"}

{"level":"info","ts":"2023-09-07T18:32:20.217Z","logger":"watcher","caller":"chains/signing.go:147","msg":"Created payload of type slsa/v1 for tekton.dev/v1beta1/TaskRun edp-delivery-sk-dev/goapp-main-build-luxw-kaniko-build","knative.dev/controller":"github.com.tektoncd.chains.pkg.reconciler.taskrun.Reconciler","knative.dev/kind":"tekton.dev.TaskRun","knative.dev/traceid":"5111d2fc-3526-443f-ae5a-a1e630098a3a","knative.dev/key":"edp-delivery-sk-dev/goapp-main-build-luxw-kaniko-build"}

{"level":"warn","ts":"2023-09-07T18:32:20.218Z","logger":"watcher","caller":"chains/signing.go:153","msg":"No signer x509 configured for tekton","knative.dev/controller":"github.com.tektoncd.chains.pkg.reconciler.taskrun.Reconciler","knative.dev/kind":"tekton.dev.TaskRun","knative.dev/traceid":"5111d2fc-3526-443f-ae5a-a1e630098a3a","knative.dev/key":"edp-delivery-sk-dev/goapp-main-build-luxw-kaniko-build"}

{"level":"info","ts":"2023-09-07T18:32:20.218Z","logger":"watcher","caller":"chains/signing.go:147","msg":"Created payload of type simplesigning for tekton.dev/v1beta1/TaskRun edp-delivery-sk-dev/goapp-main-build-luxw-kaniko-build","knative.dev/controller":"github.com.tektoncd.chains.pkg.reconciler.taskrun.Reconciler","knative.dev/kind":"tekton.dev.TaskRun","knative.dev/traceid":"5111d2fc-3526-443f-ae5a-a1e630098a3a","knative.dev/key":"edp-delivery-sk-dev/goapp-main-build-luxw-kaniko-build"}

{"level":"warn","ts":"2023-09-07T18:32:20.218Z","logger":"watcher","caller":"chains/signing.go:153","msg":"No signer x509 configured for oci","knative.dev/controller":"github.com.tektoncd.chains.pkg.reconciler.taskrun.Reconciler","knative.dev/kind":"tekton.dev.TaskRun","knative.dev/traceid":"5111d2fc-3526-443f-ae5a-a1e630098a3a","knative.dev/key":"edp-delivery-sk-dev/goapp-main-build-luxw-kaniko-build"}
```

# Steps to Reproduce the Problem
1. Use **cosign** version `2.2.0` and run: `cosign generate-key-pair k8s://tekton-chains/signing-secrets`
2. Follow the tutorial https://tekton.dev/docs/chains/getting-started-tutorial/

# Additional Info

- Kubernetes version:

  **Output of `kubectl version`:**
 ```
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.2", GitCommit:"7f6f68fdabc4df88cfea2dcf9a19b2b830f1e647", GitTreeState:"clean", BuildDate:"2023-05-17T14:20:07Z", GoVersion:"go1.20.4", Compiler:"gc", Platform:"darwin/arm64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"26+", GitVersion:"v1.26.7-eks-2d98532", GitCommit:"c9c3fc1fdf677f38b8d26568c10b921a1235fa43", GitTreeState:"clean", BuildDate:"2023-07-28T16:51:21Z", GoVersion:"go1.20.6", Compiler:"gc", Platform:"linux/amd64"}
 ```

- Tekton Pipeline version:
Client version: 0.31.2
Chains version: v0.17.0
Pipeline version: v0.51.0
Triggers version: v0.25.0
Dashboard version: v0.36.1

# No issues
No such issue when using cosign version 2.1.1 (https://github.com/sigstore/cosign/releases)

### PR DESCRIPTION
**Pull request by lcarva**
In order to support keys generated by cosign v2.2.0, we need to use, at least, that version of cosign. This also updates tektoncd/pipeline to its newest version, v0.51.0 to fix a dependency conflict with knative.dev/pkg, and go to 1.20 as required by the version of buildkite/agent/v3.

Fixes #918

# Release Notes

``` release-note
NONE
```

### CODE REVIEW

**Review by wlynch — CHANGES_REQUESTED** [review]

File: .github/workflows/kind-e2e.yaml [hunk 1]
```diff
@@ -16,11 +16,12 @@ jobs:
     strategy:
       fail-fast: false # Keep running if one leg fails.
       matrix:
+        # Keep in sync with the list of supported releases: https://kubernetes.io/releases/
         k8s-version:
-        - v1.24.x
         - v1.25.x
         - v1.26.x
         - v1.27.x
+        - v1.28.x
```
[the remark this answers is not in the document]
Comment (lcarva) [reply .github/workflows/kind-e2e.yaml]: Done. Thanks!

**Comment by lcarva** [comment]
@wlynch, this should be all set now.

**Review by wlynch — APPROVED** [review]
/lgtm

**Pull request merged**